src/FixedSizeArray.jl

```diff
@@ -2,17 +2,10 @@ export
     FixedSizeArray, FixedSizeVector, FixedSizeMatrix,
     FixedSizeArrayDefault, FixedSizeVectorDefault, FixedSizeMatrixDefault
 
-"""
-    Internal()
-
-Implementation detail. Do not use.
-"""
-struct Internal end
-
 struct FixedSizeArray{T,N,Mem<:DenseVector{T}} <: DenseArray{T,N}
     mem::Mem
     size::NTuple{N,Int}
-    function FixedSizeArray{T,N,M}(::Internal, mem::M, size::NTuple{N,Int}) where {T,N,M<:DenseVector{T}}
+    global function new_fixed_size_array(mem::M, size::NTuple{N,Int}) where {T,N,M<:DenseVector{T}}
```

Comment:
why are you putting a global function inside a constructor?

Reply:
`BigFloat` does something similar:

https://github.com/JuliaLang/julia/blob/71bfbb3b0134c05e1af49c385738b8f5d8f06e7c/base/mpfr.jl#L145-L148